### PR TITLE
Accept 'Join' messages from nodes without dc

### PR DESCRIPTION
**Pull request by raboof**
To allow a join from a 2.4 node to a 2.5.6 cluster.

### CODE REVIEW

File: akka-cluster/src/main/scala/akka/cluster/protobuf/ClusterMessageSerializer.scala [hunk 1]
```diff
@@ -37,9 +37,12 @@ class ClusterMessageSerializer(val system: ExtendedActorSystem) extends BaseSeri
     classOf[InternalClusterAction.Join] → {
       case bytes ⇒
         val m = cm.Join.parseFrom(bytes)
+        val roles = Set.empty[String] ++ m.getRolesList.asScala
         InternalClusterAction.Join(
           uniqueAddressFromProto(m.getNode),
-          Set.empty[String] ++ m.getRolesList.asScala)
+          if (roles.find(_.startsWith(ClusterSettings.DcRolePrefix)).isDefined) roles
+          else roles + (ClusterSettings.DcRolePrefix + "default")
```
Comment (johanandren) [inline akka-cluster/src/main/scala/akka/cluster/protobuf/ClusterMessageSerializer.scala]: There's a `ClusterSettings.DefaultDataCenter` constant as well

**Review by johanandren — APPROVED** [review]
LGTM, but use the constant in both places instead of a string

**Review by patriknw — APPROVED** [review]
LGTM after changing to the constant

**Review by ktoso — APPROVED** [review]
LGTM

**Pull request merged**